pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -520,3 +656,50 @@ func getSecretReferences(serviceAccount *api.ServiceAccount) sets.String {
 	}
 	return references
 }
+
+func makeServiceAccountKey(sa *api.ServiceAccount) string {
+	return fmt.Sprintf("%s/%s/%s", sa.Namespace, sa.Name, sa.UID)
+}
+
+func parseServiceAccountKey(key interface{}) ( /*ns*/ string /*name*/, string /*uid*/, types.UID, error) {
```

Comment:
I'm actually leaning toward making a struct as my key instead of munging everything into a string and then parsing it back out